File: performance.cpp
#include <iostream>
#include <chrono>

int main() {
    long long result = 0;
    int outerIterations = 10000;
    int innerIterations = 100000;

    std::cout << "Iniciando 1,000 millones de iteraciones con bucles anidados...\n";

    auto start = std::chrono::high_resolution_clock::now();

    for (int i = 0; i < outerIterations; i++) {
        for (int j = 0; j < innerIterations; j++) {
            result++;
        }
    }

    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> elapsed = end - start;

    std::cout << "\nResultado final: " << result << std::endl;
    std::cout << "Tiempo de ejecución: " << elapsed.count() << " segundos" << std::endl;

    return 0;
}

File: performance.go
package main

import (
    "fmt"
    "time"
)

func main() {
    outerIterations := 10000
    innerIterations := 100000
    result := 0

    fmt.Println("Iniciando 1,000 millones de iteraciones con bucles anidados...")

    start := time.Now()

    for i := 0; i < outerIterations; i++ {
        for j := 0; j < innerIterations; j++ {
            result++
        }
    }

    duration := time.Since(start).Seconds()

    fmt.Printf("\nResultado final: %d\n", result)
    fmt.Printf("Tiempo de ejecución: %.2f segundos\n", duration)
}
